nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/AbstractTestConversion.java

```diff
@@ -0,0 +1,396 @@
+
+        writerConfigurer = testRunner -> {
+            commonWriterConfiguration(testRunner);
+            testRunner.setProperty(writer, "Pretty Print JSON", "true");
+        };
+    }
+
+    protected void toXml(String postfix) {
+        writer = new XMLRecordSetWriter();
+        resultHandler = stringOutputHandler(getContent(postfix));
+
+        writerConfigurer = testRunner -> {
+            commonWriterConfiguration(testRunner);
+            testRunner.setProperty(writer, "pretty_print_xml", "true");
+            testRunner.setProperty(writer, "root_tag_name", "root");
+            testRunner.setProperty(writer, "record_tag_name", "nifiRecord");
+        };
+    }
+
+    protected void toAvro(String postfix) {
+        writer = new AvroRecordSetWriter();
+        resultHandler = mockFlowFile -> {
+            try {
+                ArrayList<Map<String, Object>> expected = getRecords(getByteContent(postfix));
+                ArrayList<Map<String, Object>> actual = getRecords(mockFlowFile.toByteArray());
+
+                assertEquals(expected, actual);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        };
+
+        writerConfigurer = testRunner -> {
+            commonWriterConfiguration(testRunner);
+        };
+    }
+
+    protected Consumer<TestRunner> stringInputHandler(String input) {
+        return testRunner -> testRunner.enqueue(input);
+    }
+
+    protected Consumer<TestRunner> byteInputHandler(byte[] input) {
+        return testRunner -> testRunner.enqueue(input);
+    }
+
+    protected Consumer<MockFlowFile> stringOutputHandler(String expected) {
+        return mockFlowFile -> mockFlowFile.assertContentEquals(expected);
+    }
+
+    protected String getContent(String postfix) {
+        String content = new String(getByteContent(postfix));
+
+        return content;
+    }
+
+    protected byte[] getByteContent(String postfix) {
+        byte[] content;
+        try {
+            content = Files.readAllBytes(Paths.get("src/test/resources/TestConversions/data.int_float_string." + postfix));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        return content;
+    }
+
+    protected ArrayList<Map<String, Object>> getRecords(byte[] avroData) throws IOException, MalformedRecordException {
+        RecordReader reader = new AvroReaderWithEmbeddedSchema(new ByteArrayInputStream(avroData));
+
+        return getRecords(reader);
+    }
+
+    protected ArrayList<Map<String, Object>> getRecords(RecordReader reader) throws IOException, MalformedRecordException {
+        try {
+            ArrayList<Map<String, Object>> records = new ArrayList<>();
+
+            Record record;
+            while ((record = reader.nextRecord()) != null) {
+                records.add(record.toMap());
+            }
+
+            return records;
+        } finally {
+            reader.close();
```

Comment:
Is generally bad form IMO to close a `Closeable` provided to you as an argument. It generally should be closed using a `try-with-resource` by whoever created it.

Reply:
Good point!